### ISSUE TITLE
Incompatible types for Field

### ISSUE THREAD

**Issue by zouxuoz** [final-form/react-final-form#282]
### Are you submitting a **bug report** or a **feature request**?

Bug

### What is the current behavior?

![image](https://user-images.githubusercontent.com/9217163/41718060-78484554-7564-11e8-8a0b-2c11ffc6cd45.png)

### What is the expected behavior?

Found 0 errors

### Sandbox Link

The error caused because
https://github.com/final-form/react-final-form/blob/master/src/types.js.flow#L69
conflict with
https://github.com/final-form/react-final-form/blob/master/src/types.js.flow#L81

Minimal example sandbox: https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVAXAngBwKZgDyAdgQLxgDeYOATnDgFxjECuAtgEZ61gC+Abky4CAFQRwwFanQbNOcODDwBDYvyHD8YAEJwMACylFSYAGRhxcTcGBgAJAFEAHvgDGGR7Xq1UANxVeAJhWPABGZj1DY0pUMBp6JjAwsFswLx8AQlRBdDSnVzwPDLhfAKCVELwAJkj9I2k4hLkwDFpQ1LsS2mzBIA

### What's your environment?

"final-form": "^4.8.2",
"react-final-form": "^3.6.2",

### Other

I'm locally fix that problem something like this

```jsx
export type RenderableProps<T, DOMTags> = {
  children?: ((props: T) => React.Node) | React.Node,
  component?: React.ComponentType<*> | DOMTags,
  render?: (props: T) => React.Node
}

export type FormProps = {
  subscription?: FormSubscription,
  decorators?: Decorator[]
} & Config &
  RenderableProps<FormRenderProps, string>

export type FieldProps = {
  allowNull?: boolean,
  format?: (value: any, name: string) => any,
  formatOnBlur?: boolean,
  isEqual?: (a: any, b: any) => boolean,
  name: string,
  parse?: (value: any, name: string) => any,
  subscription?: FieldSubscription,
  validate?: FieldValidator,
  validateFields?: string[],
  value?: any
} & RenderableProps<FieldRenderProps, SupportedInputs>

export type FormSpyProps = {
  onChange?: (formState: FormState) => void,
  subscription?: FormSubscription
} & RenderableProps<FormSpyRenderProps, string>
```

Can I send PR for fix this?

**Comment by erikras** [comment]
Seems like just replacing `string` with `SupportedInputs` might do it, right? After all, there are only three strings that it could really be.

**Comment by zouxuoz** [comment]
@erikras I think `Form` and `FormSpy` cannot receive `component` prop as `string`, because HTML tags with `FormSpyRenderProps` or `FormRenderProps` compatibility is not existed :)

May be something like this?

```jsx
export type RenderableProps<T> = {
  children?: ((props: T) => React.Node) | React.Node,
  render?: (props: T) => React.Node
}

export type FormProps = {
  component?: React.ComponentType<*>
  subscription?: FormSubscription,
  decorators?: Decorator[]
} & Config &
  RenderableProps<FormRenderProps>

export type FieldProps = {
  allowNull?: boolean,
  component?: React.ComponentType<*> | SupportedInputs,
  format?: (value: any, name: string) => any,
  formatOnBlur?: boolean,
  isEqual?: (a: any, b: any) => boolean,
  name: string,
  parse?: (value: any, name: string) => any,
  subscription?: FieldSubscription,
  validate?: FieldValidator,
  validateFields?: string[],
  value?: any
} & RenderableProps<FieldRenderProps>

export type FormSpyProps = {
  component?: React.ComponentType<*>,
  onChange?: (formState: FormState) => void,
  subscription?: FormSubscription
} & RenderableProps<FormSpyRenderProps>
```

**Comment by erikras** [comment]
>  I think `Form` and `FormSpy` cannot receive component prop as string, because HTML tags with `FormSpyRenderProps` or `FormRenderProps` compatibility is not existed :)

Yes! 👍  Your thinking is sound on this issue. PR me.

**Comment by erikras** [comment]
Published fix in [`v3.6.3`](https://github.com/final-form/react-final-form/releases/tag/v3.6.3).
